pkg/api/api.go

```diff
@@ -50,10 +58,25 @@ func New(tags *tags.Tags, storer storage.Storer, corsAllowedOrigins []string, lo
 	return s
 }
 
-const (
-	SwarmPinHeader = "Swarm-Pin"
-	TagHeaderUid   = "swarm-tag-uid"
-)
+func (s *server) getOrCreateTag(tagUid string) (*tags.Tag, bool, error) {
+	// if tag header is not there create a new one
+	if tagUid == "" {
+		tagName := fmt.Sprintf("unnamed_tag_%d", time.Now().Unix())
+		var err error
+		tag, err := s.Tags.Create(tagName, 0, false)
+		if err != nil {
+			return nil, false, fmt.Errorf("cannot create tag: %w", err)
+		}
+		return tag, true, nil
+	}
+	// if the tag uid header is present, then use the tag sent
+	uid, err := strconv.Atoi(tagUid)
+	if err != nil {
+		return nil, false, fmt.Errorf("cannot parse taguid: %w", err)
+	}
+	t, err := s.Tags.Get(uint32(uid))
```

Comment:
you're right; i have added a docstring to this function.

regarding the behavior: a change could be made so that if an id is supplied but the tag cannot be fetched, then create a new tag.

however, the id supplied would not match the id for the newly created tag. i think this would be somewhat confusing, so i'd vote to keep this as-is.

Reply:
https://github.com/ethersphere/bee/issues/579 :+1: